src/book/02-system/01-nubots/03-nuclear.mdx

```diff
@@ -5,3 +5,144 @@ title: NUClear
+| Last     | Instructs the PowerPlant to store the last $n$ messages (of the associated type) to the cache and provide read-only access to the subscribing reaction. `on<Last<n, Trigger<T, ...>>>()` will store the next $n$ messages of each of the `Trigger` types in a list with the oldest message first. After $n$ messages are stored, the trigger of a new reaction will cause the oldest message to be dropped and the newest copy appended to the list.                                         |
+| Optional | Is used to signify any optional properties in the DSL request. `on<Trigger<T1>, Optional<With<T2>()` will run when T1 is emitted into the system. If T2 is available, then the reaction will be given read-only access to the most recent emission of T2.                                                                                                                                                                                             |
+| Single   | Is used to specify that only one instance of the associated reaction can execute at any given point during runtime. `on<Trigger<T, ...>, Single>()` will only allow one task for this reaction to be executed or in the queue at any given time. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                                                                           |
+| Buffer   | Is used to specify that up to $n$ instances of the associated reaction can execute during runtime. `on<Trigger<T, ...>, Buffer<n>>>()` will create and schedule a new task if there are less than $n$ existing tasks for this reaction. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                                                                   |
+| Priority | Assigns a priority to any tasks created to determine scheduling order in the PowerPlant. `on<Trigger<T, ...>, Priority::HIGH>()` will schedule the task with `HIGH` priority when the reaction is triggered. Available priority settings are `REALTIME`, `HIGH`, `NORMAL`, `LOW`, and `IDLE`. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                      |
+| Sync     | Sets the synchronisation for a group of tasks. `on<Trigger<T, ...>, Sync<Group>>()` will allow only one task from `Group` to execute at any given time. This is similar to the `Single` word, but applies to multiple reactions. This word helps prevent race conditions. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                   |
```

Comment:
It's not similar to the `Single` word, you can, in fact, use both of them together. `Single` means "if this reaction is running and more data comes in drop it" and `Sync` means if something in this group is already running then queue this task until it is finished. So if you combine together you can have a setup where another reaction in the group is running, a task for this reaction is submitted, but since it has Sync it is queued. Then if another task for this reaction comes in then that second task will be dropped (because one is already queued). Single influences how things run, Sync influences how things are queued.